rd_ui/app/scripts/controllers/query_view.js

```diff
@@ -270,9 +278,19 @@
       }
       $scope.selectedTab = hash || DEFAULT_TAB;
     });
+
+    /**
+     * Access groups related
+     */
+    $scope.$watch('query.access_groups', function (newValue, oldValue) {
```

Comment:
Watch is being fired on the first check even if there was no change, so you need to make sure that `newValue !== oldValue`, otherwise it will try to save a new query.